modern/org/commcare/modern/database/TableBuilder.java

```diff
@@ -180,7 +180,7 @@ public String getTableCreateString() {
 
     //sqlite doesn't like dashes
     public static String scrubName(String input) {
-        return input.replace("-", "_");
+        return input.replace("-", "_dash_").replace(".", "_dot_").replace("+", "_plus_");
```

Comment:
Pretty sure this will require a bunch of data migrations. For example, this is used to get db column names in https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/models/database/SqlStorage.java#L152